warehouse/transformer/transformer.go

```diff
@@ -171,46 +170,20 @@ func (t *Transformer) eventContext(tec *transformEventContext) any {
 	return clonedContext
 }
 
-func (t *Transformer) addMandatoryFields(event *types.TransformerEvent) {
-	t.ensureMessageID(event)
-	t.ensureReceivedAt(event)
-}
-
-func (t *Transformer) ensureMessageID(event *types.TransformerEvent) {
-	messageID, exists := event.Message["messageId"]
-	if !exists || utils.IsBlank(messageID) {
-		event.Metadata.MessageID = "auto-" + t.uuidGenerator()
-	}
-}
-
-func (t *Transformer) ensureReceivedAt(event *types.TransformerEvent) {
-	receivedAt, exists := event.Message["receivedAt"]
-	if !exists || utils.IsBlank(receivedAt) {
-		t.setDefaultReceivedAt(event)
-		return
-	}
-
-	strReceivedAt, isString := receivedAt.(string)
-	if !isString || !utils.ValidTimestamp(strReceivedAt) {
-		t.setDefaultReceivedAt(event)
-	}
-}
-
-func (t *Transformer) setDefaultReceivedAt(event *types.TransformerEvent) {
-	if !utils.ValidTimestamp(event.Metadata.ReceivedAt) {
-		event.Metadata.ReceivedAt = t.now().Format(misc.RFC3339Milli)
-	}
-}
-
 func (t *Transformer) handleEvent(event *types.TransformerEvent, cache *cache) ([]map[string]any, error) {
 	intrOpts := extractIntrOpts(event.Metadata.DestinationType, event.Message)
 	destOpts := extractDestOpts(event.Metadata.DestinationType, event.Destination.Config)
 	jsonPathsInfo := extractJSONPathInfo(append(intrOpts.jsonPaths, destOpts.jsonPaths...))
 
 	eventType := strings.ToLower(event.Metadata.EventType)
 
+	warehouseEvent := &wtypes.WarehouseTransformerEvent{}
+	warehouseEvent.TransformerEvent = event
+	warehouseEvent.MessageID = utils.ExtractMessageID(event, t.uuidGenerator)
+	warehouseEvent.ReceivedAt = utils.ExtractReceivedAt(event, t.now)
```

Comment:
Should we encapsulate all this logic in constructor of `WarehouseTransformerEvent`?

Reply:
I am working on a separate PR to not rely on `TransformerEvent` (as we are not depending only a subset of information) and use `WarehouseTransformerEvent` with all the necessary fields required.